src/main/java/de/unijena/cheminf/fragment/fingerprint/CountFingerprint.java

```diff
@@ -0,0 +1,179 @@
+    public int getCount(int index) {
+        if (index < this.fragmentArrayOfUniqueSmiles.length && index >= 0 && this.rawMap.containsKey(index)) {
+            return this.rawMap.get(index);
+        } else if (index >= this.fragmentArrayOfUniqueSmiles.length) {
+            return this.rawMap.get(index);
+        } else {
+            return 0;
+        }
+    }
+    //
+    /**
+     * Returns the hash corresponding to the given index in the fingerprint.
+     * Since this is a key-based fingerprint, the hash value is nothing
+     * more than the position of the bin in the fingerprint (no hash value is calculated).
+     *
+     * @param index the index of the bin to return the hash for.
+     * @return the hash for the bin with the given index.
+     */
+    @Override
+    public int getHash(int index) {
+        return this.uniqueSmilesToPositionMap.get(this.fragmentArrayOfUniqueSmiles[index]);
+    }
+    //
+    /**
+     *
+     * @param fp to be merged
+     */
+    @Override
+    public void merge(ICountFingerprint fp) {
+        throw new UnsupportedOperationException();
+    }
+    //
+    /**
+     *
+     * @param behaveAsBitFingerprint
+     */
+    @Override
+    public void setBehaveAsBitFingerprint(boolean behaveAsBitFingerprint) {
+        throw new UnsupportedOperationException();
+    }
+    //
+    /**
+     * Whether the fingerprint contains the given hash.
+     *The parameter hash is not a calculated hash value, but also corresponds to the position of the bin in the fingerprint.
+     * @see    #getHash(int)
+     * For hash greater than or equals fingerprint size the method returns false and for all hash less than fingerprint size a true is returned.
+     *
+     * @param hash
+     * @return true if the fingerprint contains the given hash, otherwise false.
+     */
+    @Override
+    public boolean hasHash(int hash) {
+        if(hash >= this.fragmentArrayOfUniqueSmiles.length){
+            return false;
+        } else {
+            return this.uniqueSmilesToPositionMap.containsKey(this.fragmentArrayOfUniqueSmiles[hash]);
+        }
+    }
+    //
+    /**
+     * Get the number of times a certain hash exists in the fingerprint.
+     * Since the fragment fingerprint is a key-based fingerprint and the hash value therefore indicates the
+     * position of the respective bin in the fingerprint, each hash value occurs only once in the fingerprint.
+     * Therefore, the method returns the frequency of the fragment in the specified bin.
+     *
+     * @param hash the index of the bin to return the number of hits for.
+     * @return the number associated with the given hash
+     */
+    @Override
+    public int getCountForHash(int hash) {
+        if (hash < this.fragmentArrayOfUniqueSmiles.length && hash >= 0 && this.rawMap.containsKey(hash)) {
+            return this.rawMap.get(hash);
+        } else if (hash >= this.fragmentArrayOfUniqueSmiles.length) {
+            return this.rawMap.get(hash);
```

Comment:
Why the "else if" clause? 
"hash >= 0 && this.rawMap.containsKey(hash)" shouldn't that be enough? Then you can remove the  "else if"